internal/resources/policy/type/security/resource_security_policy.go

```diff
@@ -0,0 +1,449 @@
+							FullName: scopedFullname.fullnameClusterGroup,
+							Meta:     meta,
+							Spec:     spec,
+						},
+					}
+
+					_, err := config.TMCConnection.ClusterGroupPolicyResourceService.ManageV1alpha1ClustergroupPolicyResourceServiceUpdate(securityPolicyReq)
+					if err != nil {
+						return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control security policy entry, name : %s", securityPolicyName))
+					}
+				}
+			case organizationScope:
+				if scopedFullname.fullnameOrganization != nil {
+					securityPolicyReq := &policyorganizationmodel.VmwareTanzuManageV1alpha1OrganizationPolicyPolicyRequest{
+						Policy: &policyorganizationmodel.VmwareTanzuManageV1alpha1OrganizationPolicyPolicy{
+							FullName: scopedFullname.fullnameOrganization,
+							Meta:     meta,
+							Spec:     spec,
+						},
+					}
+
+					_, err := config.TMCConnection.OrganizationPolicyResourceService.ManageV1alpha1OrganizationPolicyResourceServiceUpdate(securityPolicyReq)
+					if err != nil {
+						return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control security policy entry, name : %s", securityPolicyName))
+					}
+				}
+			}
+
+			log.Printf("[INFO] security policy update successful")
+		}
+	}
+
+	return resourceSecurityPolicyRead(ctx, d, m)
+}
+
+func updateCheckForMeta(d *schema.ResourceData, meta *objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta) bool {
+	if !common.HasMetaChanged(d) {
+		return false
+	}
+
+	objectMeta := common.ConstructMeta(d)
+
+	if value, ok := meta.Labels[common.CreatorLabelKey]; ok {
+		objectMeta.Labels[common.CreatorLabelKey] = value
+	}
+
+	meta.Labels = objectMeta.Labels
+	meta.Description = objectMeta.Description
+
+	log.Printf("[INFO] updating security policy meta data")
+
+	return true
+}
+
+func updateCheckForSpec(d *schema.ResourceData, spec *policymodel.VmwareTanzuManageV1alpha1CommonPolicySpec) bool {
+	if !hasSpecChanged(d) {
+		return false
+	}
+
+	policySpec := constructSpec(d)
+
+	spec.Input = policySpec.Input
+	spec.NamespaceSelector = policySpec.NamespaceSelector
+	spec.Recipe = policySpec.Recipe
+	spec.RecipeVersion = policySpec.RecipeVersion
+
+	log.Printf("[INFO] updating security policy spec")
+
+	return true
+}
+
+func resourceSecurityPolicyDelete(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	securityPolicyName, _ := d.Get(nameKey).(string)
+
+	scopedFullname := constructScope(d, securityPolicyName)
+
+	if scopedFullname != nil {
+		switch scopedFullname.scope {
+		case clusterScope:
+			if scopedFullname.fullnameCluster != nil {
+				err := config.TMCConnection.ClusterPolicyResourceService.ManageV1alpha1ClusterPolicyResourceServiceDelete(scopedFullname.fullnameCluster)
+				if err != nil && !clienterrors.IsNotFoundError(err) {
+					return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control security policy entry, name : %s", securityPolicyName))
```

Comment:
Should we have more specific error messages pertaining to each scope here as well as in create and update?

Reply:
I can add the scope information, but I have followed the convention followed for other resources.

@shreyassreenivas and @shobha2626, any thoughts?

Reply:
Done @Vasundharashukla , PTAL